Sources/AWSLogging/CloudwatchJsonStandardErrorLoggerV2.swift

```diff
@@ -0,0 +1,188 @@
+    // is called.
+    public func run() async {
+        for await jsonMessage in self.entryStream {
+            var stream = self.stream
+            stream.write("\(jsonMessage)\n")
+        }
+    }
+    
+    public subscript(metadataKey metadataKey: String) -> Logger.Metadata.Value? {
+        get {
+            return metadata[metadataKey]
+        }
+        set {
+            metadata[metadataKey] = newValue
+        }
+    }
+    
+    /**
+     Set the logger implementation of the LoggerAPI to this type.
+     */
+    public static func enableLogging(minimumLogLevel: Logger.Level = .info,
+                                     metadataTypes: [String: MetadataType] = [:]) -> CloudwatchJsonStandardErrorLoggerV2 {
+        let logger = CloudwatchJsonStandardErrorLoggerV2(minimumLogLevel: minimumLogLevel, metadataTypes: metadataTypes)
+        
+        LoggingSystem.bootstrap { label in
+            return logger
+        }
+        
+        return logger
+    }
+    
+    public func log(level: Logger.Level, message: Logger.Message,
+                    metadata: Logger.Metadata?, file: String, function: String, line: UInt) {
+        let shortFileName: String
+        if let range = file.range(of: "Sources/") {
+            let startIndex = file.index(range.lowerBound, offsetBy: sourcesSubString.count)
+            shortFileName = String(file[startIndex...])
+        } else {
+            shortFileName = file
+        }
+        
+        let metadataToUse: Logger.Metadata
+        if let metadata = metadata {
+            metadataToUse = self.metadata.merging(metadata) { (global, local) in local }
+        } else {
+            metadataToUse = self.metadata
+        }
+        
+        var codableMetadata: [String: String] = [:]
+        var codableMetadataInts: [String: Int] = [:]
+        metadataToUse.forEach { (key, value) in
+            // determine the metadata type
+            let metadataType: MetadataType
+            if let theMetadataType = self.metadataTypes[key] {
+                metadataType = theMetadataType
+            } else {
+                metadataType = .String
+            }
+            
+            // add to the appropriate dictionary, converting if necessary
+            switch metadataType {
+            case .String:
+                codableMetadata[key] = value.description
+            case .Int:
+                codableMetadataInts[key] = Int(value.description)
+            }
+        }
+        
+        codableMetadata["fileName"] = shortFileName
+        codableMetadataInts["line"] = Int(line)
+        codableMetadata["function"] = function
+        codableMetadata["level"] = level.rawValue
+        codableMetadata["message"] = "\(message)"
+        
+        // pass to the global dispatch queue for serialization
+        // schedule at a low priority to avoid disrupting request handling
+        DispatchQueue.global().async(qos: .utility) {
```

Comment:
Just for my understanding, why do we need to use a `DispatchQueue` here instead of `Task` or some other Concurrency primitive?

Reply:
It appears to give better performance to serialise on an unbounded thread pool.